### PR TITLE
AZ IOT HUB apply-configuration needs hubname

### LINKED ISSUE #167: AZ IOT HUB apply-configuration needs hubname.
If user has old version of az cli iot extension installed they get this: 

`az iot hub apply-configuration: error: argument --hub-name/-n is required
`

- add the -n parameter to the apply-configuration call.  you can get it in IOTHUB_CONNECTION_INFO.HostName

apply-configuration might need ONLY hubname, but HostName has [name].azuredevices.net.

Therefore, You might have to split the ConnectionString.HostName property and add a new property to that class called HubName.

### PR DESCRIPTION
**Pull request by kln4dt**
this commit fixes #167

### CODE REVIEW

File: iotedgedev/azurecli.py [hunk 1]
```diff
@@ -226,10 +226,10 @@ def list_resource_groups(self):
 
         return result
 
-    def apply_configuration(self, deviceId, connection_string, config):
+    def apply_configuration(self, deviceId, connection_string, hubName, config):
```
Comment (jonbgallant) [inline iotedgedev/azurecli.py]: I just noticed that the parameter names here are a mix of camelCase and underscores.  Can you make consistent with underscores?

`(self, device_id, connection_string, hub_name, config)`

File: iotedgedev/edge.py [hunk 2]
```diff
@@ -14,7 +14,7 @@ def deploy(self):
         self.envvars.verify_envvar_has_val("DEVICE_CONNECTION_STRING", self.envvars.DEVICE_CONNECTION_STRING)
         self.envvars.verify_envvar_has_val("DEPLOYMENT_CONFIG_FILE", self.envvars.DEPLOYMENT_CONFIG_FILE)
 
-        self.azure_cli.apply_configuration(self.envvars.DEVICE_CONNECTION_INFO.DeviceId, self.envvars.IOTHUB_CONNECTION_STRING, self.envvars.DEPLOYMENT_CONFIG_FILE_PATH)
+        self.azure_cli.apply_configuration(self.envvars.DEVICE_CONNECTION_INFO.DeviceId, self.envvars.IOTHUB_CONNECTION_STRING, self.envvars.IOTHUB_CONNECTION_INFO.HubName, self.envvars.DEPLOYMENT_CONFIG_FILE_PATH)
```
Comment (jonbgallant) [inline iotedgedev/edge.py]: 1. Change the check above to IOTHUB_CONNECTION_INFO
`self.envvars.verify_envvar_has_val("IOTHUB_CONNECTION_INFO", self.envvars.IOTHUB_CONNECTION_INFO)`

2. Change following to use IOTHUB_CONNECTION_INFO

`self.azure_cli.apply_configuration(self.envvars.DEVICE_CONNECTION_INFO.DeviceId, self.envvars.IOTHUB_CONNECTION_INFO.ConnectionString, self.envvars.IOTHUB_CONNECTION_INFO.HubName, self.envvars.DEPLOYMENT_CONFIG_FILE_PATH)`

File: iotedgedev/connectionstring.py [hunk 3]
```diff
@@ -13,6 +13,8 @@ def __init__(self, value):
 
             if self.data:
                 self.HostName = self["hostname"]
+                if self.HostName:
```
Comment (jonbgallant) [inline iotedgedev/connectionstring.py]: Keep `def __init__(self, value)` as is.

Change self.value to self.ConnectionString

`self.ConnectionString = value`

And then update all references to self.value to self.ConnectionString.

Also, line 7 above references value.split(';'), that should be self.ConnectionString.split(';')

File: iotedgedev/edge.py (hunk 2 above)
Comment (kln4dt) [reply iotedgedev/edge.py]: shouldn't the check for device connection string be changed to DEVICE_CONNECTION_INFO too?

Comment (jonbgallant) [reply iotedgedev/edge.py]: Yes.

**Pull request merged**